.gitlab-ci.yml

```diff
@@ -0,0 +1,138 @@
+stages:
+  - install_deps
+  - lint
+  - test
+  - build
+  - deploy
+
+variables:                         &default-vars
+  CI_IMAGE:                        "node:16.10"
+  APP_PORT:                        3000
+  DEPLOYMENT_IMAGE_REGISTRY:       docker.io/paritytech/pr-custom-review
+  DEPLOYMENT_TAG:                  $CI_COMMIT_SHORT_SHA-$CI_PIPELINE_ID
+  # Vault Section
+  VAULT_ADDR:                      https://vault.parity-mgmt-vault.parity.io
+  VAULT_AUTH_PATH:                 gitlab-parity-io-jwt
+  VAULT_AUTH_ROLE:                 cicd_gitlab_parity_${CI_PROJECT_NAME}
+  HELM_SECRETS_DRIVER:             vals
+
+default:
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - kubernetes-parity-build
+  image:                           $CI_IMAGE
+
+.common-refs:                      &common-refs
+  rules:
+    - if: $CI_COMMIT_REF_NAME == "master"
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                 # PRs
+
+.staging-refs:                     &staging-refs
+  rules:
+    - if: $CI_COMMIT_TAG =~ /^stg-v[0-9]+\.[0-9]+.*$/  # e.g. stg-v0.1
+
+.production-refs:                  &production-refs
+  rules:
+    - if: $CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+.*$/      # e.g. v0.1
+
+install-deps:
+  <<:                              *common-refs
+  stage:                           install_deps
+  script:
+    - yarn
+  artifacts:
+    name:                          install-deps_$CI_PIPELINE_ID
+    when:                          on_success
+    expire_in:                     1 hour
+    paths:
+      - ./node_modules
+
+lint:
+  <<:                              *common-refs
+  stage:                           lint
+  script:
+    # TODO: Use pre-commit run --all
+    - echo "lint"
+
+security-scan:
+  <<:                              *common-refs
+  stage:                           test
+  image:                           aquasec/trivy:latest
+  script:
+    - trivy fs .
```

Comment:
security-scan is failing with `503 Service Unavailable` https://gitlab.parity.io/parity/mirrors/pr-custom-review/-/jobs/1581967#L33

I think the job is implemented correctly, so no code changes needed here.

I think we can disable it and create a ticket to re-enable it after this PR is merged.

Reply:
Looks like it was temporary issue. Now it's working https://gitlab.parity.io/parity/mirrors/pr-custom-review/-/jobs/1582086